tls/s2n_signature_algorithms.c

```diff
@@ -76,8 +76,6 @@ static S2N_RESULT s2n_signature_scheme_validate_for_recv(struct s2n_connection *
     if (conn->actual_protocol_version >= S2N_TLS13) {
         RESULT_ENSURE_NE(scheme->hash_alg, S2N_HASH_SHA1);
         RESULT_ENSURE_NE(scheme->sig_alg, S2N_SIGNATURE_RSA);
-    } else {
-        RESULT_ENSURE_NE(scheme->sig_alg, S2N_SIGNATURE_RSA_PSS_PSS);
```

Comment:
I'm surprised there wasn't any logic needed in order to gate RSA-PSS certs behind a non-RSA key exchange. How is this selection implemented? Is it worth testing this at all?

Reply:
I guess it is pretty surprising, but [that logic already exists](https://github.com/aws/s2n-tls/blob/80003e672fb722e08421bf6e66db51dff20b564a/tls/s2n_auth_selection.c#L67-L77), and [is already tested](https://github.com/aws/s2n-tls/blob/main/tests/unit/s2n_auth_selection_test.c#L248-L269). I'm thinking when we wrote the selection logic we weren't thinking about what was and wasn't allowed per protocol, but what was theoretically possible.

Reply:
Wow, nice! It was implemented with a lot of forethought I guess.